ximage/tests/labels/test_labels.py

```diff
@@ -0,0 +1,733 @@
+    dataset = xr.Dataset({variable_name: data_array})
+    dataset_returned = labels.label(dataset, variable_name)
+    labels_array_expected = np.array(
+        [
+            [3, 3, _, _, 2],
+            [_, _, _, _, 2],
+            [_, _, _, _, _],
+            [1, _, _, _, _],
+            [1, 1, _, 4, _],
+        ]
+    )
+    assert np.array_equal(dataset_returned.coords["label"], labels_array_expected, equal_nan=True)
+
+
+def test_highlight_label():
+    """Test highlight_label"""
+
+    _ = float("nan")
+    labels_array = xr.DataArray(
+        [
+            [_, 1, 2],
+            [3, _, 2],
+            [3, 3, _],
+        ]
+    )
+    label_name = "test_label"
+    label_id = 2
+    dataset = xr.Dataset({label_name: labels_array})
+    dataset_returned = labels.highlight_label(dataset, label_name, label_id)
+    labels_array_expected = np.array(
+        [
+            [0, 0, 2],
+            [0, 0, 2],
+            [0, 0, 0],
+        ]
+    )
+    assert np.array_equal(dataset_returned[label_name], labels_array_expected)
+
+
+# Tests for internal functions #################################################
+
+
+def test_mask_buffer():
+    """Test mask dilation"""
+
+    mask_ini = np.array(
+        [
+            [0, 0, 0],
+            [0, 1, 0],
+            [0, 0, 0],
+        ],
+        dtype=bool,
+    )
+
+    footprint = np.array(
+        [
+            [0, 1, 0],
+            [1, 1, 1],
+            [0, 1, 0],
+        ],
+        dtype=bool,
+    )
+
+    mask_dilated_target = footprint
+
+    # No dilation
+    assert np.array_equal(labels._mask_buffer(mask_ini, None), mask_ini)
+    assert np.array_equal(labels._mask_buffer(mask_ini, 0), mask_ini)
+
+    # Dilation of 1
+    assert np.array_equal(labels._mask_buffer(mask_ini, 1), mask_dilated_target)
+
+    # Dilation with mask
+    assert np.array_equal(labels._mask_buffer(mask_ini, footprint), mask_dilated_target)
+
+    # Errors
+    with pytest.raises(TypeError):
+        labels._mask_buffer(mask_ini, "invalid type")
+
+    with pytest.raises(ValueError):
+        labels._mask_buffer(mask_ini, -1)
+
+    with pytest.raises(ValueError):
+        non_2D_footprint = np.array([1, 1, 1])
+        labels._mask_buffer(mask_ini, non_2D_footprint)
+
+
+def test_check_array():
+    """Test array dimensions checks and conversion to np.ndarray"""
+
+    # Check error if non-2D
+    with pytest.raises(ValueError):
+        labels._check_array(np.array([1, 2, 3]))
+
+    with pytest.raises(ValueError):
+        labels._check_array(np.ones((3, 3, 3)))
+
+    # Check error if dimension of size 0
+    with pytest.raises(ValueError):
+        labels._check_array(np.ones((0, 3)))
+
+    # Check conversion to np.ndarray
+    array = np.ones((3, 3))
+
+    def check(array):
+        checked_array = labels._check_array(array)
+        assert isinstance(checked_array, np.ndarray)
+        assert np.array_equal(checked_array, array)
+
+    apply_to_all_array_types(check, array)
```

Comment:
There is a small fix to be done here. 
`C:\Users\runneradmin\micromamba\envs\ximage\Lib\site-packages\dask\array\core.py:1711: FutureWarning: The `numpy.array_equal` function is not implemented by Dask array. You may want to use the da.map_blocks function or something similar to silence this warning. Your code may stop working in a future release.`